lighthouse-core/gather/gather-runner.js

```diff
@@ -594,21 +573,22 @@ class GatherRunner {
         Object.assign(artifacts, passResults.artifacts);
 
         // If we encountered a pageLoadError, don't try to keep loading the page in future passes.
-        if (passResults.pageLoadError) break;
+        if (passResults.pageLoadError) {
+          baseArtifacts.PageLoadError = passResults.pageLoadError;
+          break;
+        }
 
         if (isFirstPass) {
           await GatherRunner.populateBaseArtifacts(passContext);
           isFirstPass = false;
         }
       }
 
-      await GatherRunner.disposeDriver(driver, options);
       GatherRunner.finalizeBaseArtifacts(baseArtifacts);
       return /** @type {LH.Artifacts} */ ({...baseArtifacts, ...artifacts}); // Cast to drop Partial<>.
-    } catch (err) {
-      // cleanup on error
+    } finally {
+      // Clean up regardless of error.
       GatherRunner.disposeDriver(driver, options);
```

Comment:
I like the simplification, but now we can't await it in the success case :/

maybe it doesn't matter and we can add a comment calling out why it doesn't matter?

worth noting that it does `clearDataForOrigin` which given the recent changes in Chrome has taken multiple seconds on my machine lately :(

Reply:
> but now we can't await it in the success case

Good catch, that wasn't intentional. I wonder how bad it would be to await the error case... Presumably it was so errors in disposing didn't overwrite the originating error, which probably often go hand in hand?

Kind of ridiculous we can't express what we mean here.

I could also just change it back, it's not a huge win.

Reply:
yeah maybe just change it back with a comment so we don't forget this again next time :)

Reply:
This pattern originally popped up way back in #639 (in the course of @wardpeet adding the multiple tabs check!), though originally we didn't wait for disconnect in the failure *or* success cases.

Looking at #2359, where the successful path was integrated into the promise chain, I'm half convinced we should be doing `await GatherRunner.disposeDriver(driver, options).catch(() => {});` in the error case to avoid ever exiting before Chrome is able to exit, but I'm not able to induce a problem, so leaving it as it was :)

Reply:
I think I'm with you on `.catch` even if we don't `await` it, but w/e :)